src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -678,10 +692,14 @@ xocl_read_axlf_helper(struct xocl_dev *xdev, struct drm_xocl_axlf *axlf_ptr)
 done:
 	if (size < 0)
 		err = size;
-	if (err != 0) {
+	/*
+	 * Always give up ownership for multi process use case; the real locking
```

Comment:
If we always unlock xclbin, why do we lock it in the first place?

Reply:
This is to keep the userpf population of xclbin metadata in sync with image running on FPGA (actually loaded by mgmt). Once the user gives up the lock, another process can change the image on FPGA via mgmt and will be then responsible to sync the view in userpf -- in the current architecture.

Reply:
Why do we need to keep them in sync? Does userpf interact with the xclbin on the FPGA while populating metadata somehow? It seems not...

Reply:
This is incremental change from before where this lock would be given up only at process exit. Now we want to give up the lock right after we have updated the metadata. 2019.1 let's revisit this.

Reply:
> This is incremental change from before where this lock would be given up only at process exit. Now we want to give up the lock right after we have updated the metadata. 2019.1 let's revisit this.

Fair enough, yes.